lib/bozo/test_runners/nunit.rb

```diff
@@ -101,7 +109,29 @@ def runner_args
     end
     
     def execute
-      execute_command :nunit, [runner_path] << runner_args
+      if @execute_in_parallel
+        failed_projects = []
+        threads = []
+
+        @projects.each do |project|
+          t = Thread.new {
+            begin
+              execute_command :nunit, [runner_path] << runner_args([project], "#{project}-#{Time.now.to_i}")
+            rescue
+              failed_projects << project
```

Comment:
Good spot, [`Queue`](http://ruby-doc.org/stdlib-2.0.0/libdoc/thread/rdoc/Queue.html#method-i-push) can handle the required locking. Getting it into an `Array` after is a little fiddly:

```ruby
arr = []

while !queue.empty?
  arr << queue.pop
end 
```